### PR TITLE
Fix 51534: spot instance request stay opened when module exit with timeout

### PR DESCRIPTION
**Pull request by vutkin**
##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

ec2 module
##### ADDITIONAL INFORMATION

related issue: https://github.com/ansible/ansible/issues/51534

```paste below

```

### CODE REVIEW

**Comment by vutkin** [comment]
Hi @lwade @tgerla, could you kindly review a PR?

**Comment by tgadiev** [comment]
@gundalow  can you take a look?

**Review by willthames — APPROVED** [review]

**Comment by willthames** [comment]
This looks good to me. If you are likely to want this to be backported, you'll need to add a [changelog fragment](https://docs.ansible.com/ansible/latest/reference_appendices/release_and_maintenance.html#changelogs-how-to)

File: changelogs/fragments/51535-ec2-fix-spot-request-end-date.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+bugfixes:
+  - ec2 - Correctly sets the end date of the Spot Instance request. Sets `ValidUntil` value in proper way so it will be auto-canceled through `spot_wait_timeout` interval.
+
```
Comment (Akasurde) [inline changelogs/fragments/51535-ec2-fix-spot-request-end-date.yaml]: You have extra line over here.

**Comment by Akasurde** [comment]
@vutkin ping me once you are done with this, I will merge it.

**Comment by vutkin** [comment]
Hi @Akasurde , could you please merge PR?

**Comment by Akasurde** [comment]
@vutkin Thanks for the contribution. @willthames Thanks for the review.

**Comment by tgadiev** [comment]
@Akasurde how about backporting this fix to current versions?

**Comment by Akasurde** [comment]
Backport 2.7 = https://github.com/ansible/ansible/pull/51922

**Pull request merged**